controls/V-72023.rb

```diff
@@ -48,26 +41,27 @@
 -rw-r--r-- 1 smithj smithj 193 Mar  5 17:06 file2
 -rw-r--r-- 1 smithj smithj 231 Mar  5 17:06 file3
 
-If any files are found with an owner different than the home directory user, this is
-a finding."
-  tag "fix": "Change the owner of a local interactive user’s files and directories
-to that owner. To change the owner of a local interactive user’s files and
-directories, use the following command:
+If any files are found with an owner different than the home directory user,
+this is a finding."
+  tag "fix": "Change the owner of a local interactive user’s files and
+directories to that owner. To change the owner of a local interactive user’s
+files and directories, use the following command:
 
 Note: The example will be for the user smithj, who has a home directory of
 \"/home/smithj\".
 
 # chown smithj /home/smithj/<file or directory>"
+  tag "fix_id": "F-78375r1_fix"
 
-  # Assumption - users' home directories created in "home"
-  home_dirs = command('ls -d /home/*').stdout.split("\n")
-  home_dirs.each do |home|
-    home_files = command("find #{home} ! -name '.*'").stdout.split("\n")
-    home_user = home.split("/")
-    home_files.each do |curr_file|
-      describe file(curr_file) do
-        its('owner') { should cmp "#{home_user[2]}" }
-      end
-    end
+  IGNORE_SHELLS = NON_INTERACTIVE_SHELLS.join('|')
+
+  findings = Set[]
+  users.where{ !shell.match(IGNORE_SHELLS) && (uid >= 1000 || uid == 0)}.entries.each do |user_info|
+    next if EXEMPT_HOME_USERS.include?("#{user_info.username}")
+    findings = findings + command("find #{user_info.home} -not -user #{user_info.username}").stdout.split("\n")
```

Comment:
Global finds should be marked as dangerous and should always use `-xdev -xautofs` at a minimum